File: server/src/main/java/by/brawl/entity/Hero.java
package by.brawl.entity;

import by.brawl.entity.bodypart.Bodypart;
import by.brawl.entity.bodypart.Equipment;

import javax.persistence.*;
import java.util.LinkedHashSet;
import java.util.Set;

@Entity
public class Hero extends NamedEntity {

    @ManyToOne
    private Account owner;

    @ManyToMany
    @JoinTable(name = "hero_spell",
            joinColumns = @JoinColumn(name = "hero_id"),
            inverseJoinColumns= @JoinColumn(name = "spell_id"))
    private Set<Spell> spells = new LinkedHashSet<>();

    @ManyToMany
    @JoinTable(name = "hero_bodypart",
            joinColumns = @JoinColumn(name = "hero_id"),
            inverseJoinColumns = @JoinColumn(name = "bodypart_id"))
    private Set<Bodypart> bodyparts = new LinkedHashSet<>();

    @ManyToMany
    @JoinTable(name = "hero_equipment",
            joinColumns = @JoinColumn(name = "hero_id"),
            inverseJoinColumns = @JoinColumn(name = "equipment_id"))
    private Set<Equipment> equipments = new LinkedHashSet<>();

    public Account getOwner() {
        return owner;
    }

    public Set<Spell> getSpells() {
        return spells;
    }

    public Set<Bodypart> getBodyparts() {
        return bodyparts;
    }

    public Set<Equipment> getEquipments() {
        return equipments;
    }

    @Override
    public String toString() {
        return "Hero{" +
                "name='" + getName() + '\'' +
                ", owner=" + owner.getUsername() +
                '}';
    }
}

File: server/src/main/java/by/brawl/entity/Hero.kt
package by.brawl.entity

import by.brawl.entity.bodypart.Bodypart
import by.brawl.entity.bodypart.Equipment
import java.util.*
import javax.persistence.*

@Entity
class Hero(id: String?,
           name: String,
           @ManyToOne val owner: Account,

           @ManyToMany
           @JoinTable(
                   name = "hero_spell",
                   joinColumns = arrayOf(JoinColumn(name = "hero_id")),
                   inverseJoinColumns = arrayOf(JoinColumn(name = "spell_id")))
           val spells: Set<Spell> = LinkedHashSet(),

           @ManyToMany
           @JoinTable(
                   name = "hero_bodypart",
                   joinColumns = arrayOf(JoinColumn(name = "hero_id")),
                   inverseJoinColumns = arrayOf(JoinColumn(name = "bodypart_id")))
           val bodyparts: Set<Bodypart> = LinkedHashSet(),

           @ManyToMany
           @JoinTable(
                   name = "hero_equipment",
                   joinColumns = arrayOf(JoinColumn(name = "hero_id")),
                   inverseJoinColumns = arrayOf(JoinColumn(name = "equipment_id")))
           val equipments: Set<Equipment> = LinkedHashSet())
    : NamedEntity(id, name) {

    private constructor(): this(id = null, name = "Default", owner = Account())

    override fun toString(): String {
        return "Hero(owner=$owner, spells=$spells, bodyparts=$bodyparts, equipments=$equipments)"
    }

    companion object {

        fun from(name: String,
                 owner: Account,
                 spells: Set<Spell>,
                 bodyparts: Set<Bodypart>,
                 id: String? = null): Hero {
            return Hero(id, name, owner, spells, bodyparts)
        }
    }
}